Examples/c++/Linked List/DLList.h

```diff
@@ -0,0 +1,401 @@
+#pragma once
+
+#include <algorithm>
+#include <utility>
+
+#ifdef DEBUG
+	#define ASSERT(x,...) { if(!(x)){ std::cout << __VA_ARGS__ << std::endl; __debugbreak(); } }
+#else
+	#define ASSERT(x,...)
+#endif
+
+namespace DSA
+{
+	template<typename List>
+	class DLLIterator
+	{
+	public:
+		using ValueType = typename List::ValueType;
+		using NodePtr = typename List::NodePtr;
+		using PointerType = ValueType*;
+		using RefType = ValueType&;
+	public:
+		DLLIterator()
+			: m_Ptr(nullptr), m_SentinelNode(nullptr)
```

Comment:
why naming with a preffix and underscore? everything above is named cammelCase

Reply:
I use m_ to mark my member/private variables only. The names of the variables will be changed to be more sensible and discribing.